draco/process_weights.py

```diff
@@ -0,0 +1,76 @@
+import re
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Dict
+
+from draco.asp_utils import get_constants
+
+asp_path = Path(__file__).resolve().parent / "asp"
+
+
+@dataclass(frozen=True)
+class WeightsAssign:
+    """Class for weights and assigning weights in an Answer Set Programming (ASP) program.
+
+    Attributes:
+        :weights_program: The weights constants in the program.
+        :assign_program: The weight assigning functions in the program.
+        :weights: The weight constants as a dictionaty.
+    """
+
+    weights_program: str
+    assign_program: str
+    weights: Dict[str, int]
+
+
+def get_weights_assigned(weight_path: Path, assign_path: Path) -> WeightsAssign:
+    """Reads the weights file and generates assign_weights.lp"""
+    with open(weight_path) as weight_constants:
+        const_prog = weight_constants.read()
+        const_dict = get_constants(const_prog)
+        assign_prog = "%% GENERATED FILE. DO NOT EDIT.\n"
+
+        for name in const_dict:
+            match = re.search("(.*)_weight", name)
+            if match:
+                assign_prog += f"preference_weight({match.group(1)},{name}).\n"
+
+    with open(assign_path, "w") as assign:
+        assign.write(assign_prog)
+
+    return WeightsAssign(const_prog, assign_prog, const_dict)
+
+
+def set_new_weights(
```

Comment:
I don't think we should have this method. The problem with writing the weights file is that if someone does this when they import the library, the next time they use the library, the weights will be different. This breaks a fundamental assumption that libraries are stateless.

Reply:
removed it. The reason I added it was that we already made it possible to change the constant values temporarily in `run_clingo`. Besides, the users are also able to directly modify the .lp files including weights.lp, so I thought we could provide a method for that.